### PR TITLE
Support for SmartMeasure SM300D2-VO2 air quality multi-sensor

### PR DESCRIPTION
**Pull request by xoseperez**
The SmartMeasure SM300D2-VO2 (https://es.aliexpress.com/item/32984571140.html) is a multi-sensor device that monitors CO2, Formaldehyde, TVOC, PM 2.5 and 10, temperature and humidity. Data is sent via UART (only RX).

### CODE REVIEW

File: code/espurna/sensors/SM300D2Sensor.h [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            }
+        }
+
+        // ---------------------------------------------------------------------
+
+        void _parse() {
+
+            #if SENSOR_DEBUG
+            char tmp[50];
+            for (unsigned char i=0; i<17; i++) sprintf(&tmp[2*i], "%02X", _buffer[i]);
```
Comment (mcspr) [inline code/espurna/sensors/SM300D2Sensor.h]: just to note, utils.h have hex{Encode,Decode} that can handle the whole range in one call

File: code/espurna/sensors/SM300D2Sensor.h [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            _tvoc = 256 * _buffer[6] + _buffer[7];
+
+            // PM 2.5
+            _pm25 = 256 * _buffer[8] + _buffer[9];
+
+            // PM 10
+            _pm100 = 256 * _buffer[10] + _buffer[11];
+
+            // Temperature
+            _temperature = _buffer[12] + (float) _buffer[13] / 10.0;
```
Comment (mcspr) [inline code/espurna/sensors/SM300D2Sensor.h]: buffer[12] is signed btw

ref. aliexpress picture-doc
> when temperature data bit7 of b13 = 1, it means negative temperature, and when bit7 = 0 of b13, it means positive temperature. For example, when b13 = 9Bh, at this time bit7 = 1, it means negative temperature, and the actual temperature is -27 C; if B13 = 1Bh, then bit7 = 0, it means positive temperature, then the actual temperature is 27 C

File: README.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ ESPurna ("spark" in Catalan) is a custom firmware for ESP8285/ESP8266 based smar
 It uses the Arduino Core for ESP8266 framework and a number of 3rd party libraries.
 
 [![version](https://img.shields.io/badge/version-1.15.0--dev-brightgreen.svg)](CHANGELOG.md)
-[![branch](https://img.shields.io/badge/branch-dev-orange.svg)](https://github.com/xoseperez/espurna/tree/dev/)
+[![branch](https://img.shields.io/badge/branch-sm300d2-orange.svg)](https://github.com/xoseperez/espurna/tree/sm300d2/)
```
Comment (mcspr) [inline README.md]: (todo) will need to edit before merge

**Review by mcspr — APPROVED** [review]

File: code/espurna/sensors/SM300D2Sensor.h (hunk 1 above)
Comment (xoseperez) [reply code/espurna/sensors/SM300D2Sensor.h]: nice!

File: code/espurna/sensors/SM300D2Sensor.h (hunk 2 above)
Comment (xoseperez) [reply code/espurna/sensors/SM300D2Sensor.h]: You are right, will fix it.

**Pull request merged**